taccsite_cms/urls.py

```diff
@@ -6,42 +6,37 @@
 from django.conf.urls.static import static
 from django.conf.urls import include, url
 from django.contrib import admin
+from django.contrib.auth import views
 from django.contrib.sitemaps.views import sitemap
 from django.contrib.staticfiles.urls import staticfiles_urlpatterns
 from django.views.static import serve
+from taccsite_cms import remote_cms_auth as remote_cms_auth
 
 from django.http import request
 from django.views.generic.base import RedirectView
-
 admin.autodiscover()
 
 urlpatterns = [
     url(r'^sitemap\.xml$', sitemap,
         {'sitemaps': {'cmspages': CMSSitemap}}),
 
     url(r'^admin/', admin.site.urls),  # NOQA
+    url(r'^cms/logout/', views.LogoutView.as_view(), name='logout'),
 ]
 
-if settings.PORTAL:
+if getattr(settings, 'INCLUDES_CORE_PORTAL', True):
     from django.views.generic.base import TemplateView
     urlpatterns += [
         # FAQ: Allow direct access to markup for Portal and User Guide to render
         url(r'^cms/nav/search/markup/$', TemplateView.as_view(template_name='nav_search.raw.html'), name='search_bar_markup'),
         url(r'^cms/nav/pages/markup/$', TemplateView.as_view(template_name='nav_cms.raw.html'), name='menu_pages_markup'),
         url(r'^cms/header/branding/markup/$', TemplateView.as_view(template_name='header_branding.html'), name='header_branding_markup'),
         url(r'^cms/header/logo/markup/$', TemplateView.as_view(template_name='header_logo.html'), name='header_logo_markup'),
-    ]
-
-if settings.FEATURES['blog']:
-    urlpatterns += [
-        # Support `taggit_autosuggest` (from `djangocms-blog`)
-        url(r'^taggit_autosuggest/', include('taggit_autosuggest.urls')),
```

Comment:
Good question, thanks for asking how we can solve this. 

This actually affects the way the whole blog is set-up. Since this base repo is meant to support a number of CMS's (most of which won't need a blog) there are a couple of ways we can add custom functionality like this without using a lot of conditional logic in the base project. One option is to break out the blog into an app in it's own repo (I believe the orignal code for this is set-up that way so you can potentially fork that repo [or just reference that project in requirements and override what you need] and follow that same pattern) and then either publish the app to pypi and add a reference in requirements.txt or point to the repo directly in the requirements file. You can add the required urls.py changes in the taccsite_custom directory for any site that may (including INSTALL_APPS entries, etc) need the blog. 

Alternatively, if you don't want to create a new repo or think the blog code will only ever serve one site, you could move the code to the specific taccsite_custom dir and  make the changes there.

Personally, I think the first option is a great opportunity to fork a popular project and get something published to pypi that others may be able to use. 

Either way, the idea is to find solutions that centralize as little code as possible, and push out deviations from core so that features are configurable via deltas in each custom cms project instead of the core project having to be aware of every possible feature used in a child/custom implementation.

thoughts?

Reply:
I would be in favor of allowing this logic to remain in the CMS code under `urls` because this is a key idea that Wes designed features around to allow the CMS to dynamically grow as needed rather than having to over-engineer a solution up front that fits all sdizes (when those sizes are forever unknown and TBD). To remove this simple bit of logic now for the sake of unneeded (and unexpected) austerity is counter to the work he has been doing on the CMS.

Explanation:

When we split the CMS codebase off a few years back, we intended to let the CMS codebase evolve to handle all of the customization needs that CMD will encounter for site content across the various sites. We broke it apart from the Core-Portal so that it actually can inculcate all these kinds of edge case features. We can not know in advance what sites will want the blog feature (or future features built around the CMS) but we know at least one does and several probably will, given the design options that feature enables for CMD.

Reply:
<details><summary>Known CMS Usage of Blog: Texascale, TACC</summary>

- Texascale should (installed, unused but enabled for testing).
- TACC CMS will

</details>

<details><summary>Installable App for Blog: Cool, but likely not now.</summary>

- Cool. This isolates and commits support for the feature.
- But, this endeavor is too large, for me, given my current skill to do in a week.
- Side note: I don't think TACC customization will serve the community. But PRs to `djangocms-blog` could.

</details>

<details><summary>Within <code>taccsite_custom</code>: Feasible.</summary>

It would be "saving" code, not using it—just like the other settings stored in there. Unless, a `settings.custom` file in that repo's directories are intended to be applied with the upcoming architecture change.

</details>

<details><summary>Use <code>settings_custom.example.py</code>: I propose this.</summary>

Portal has this file. It supports saving example customization and documentation for them. If blog becomes worth having more than just these settings and a `urls.py` bit, we can explore an "Installable App for Blog".

</details>

Reply:
@taoteg 
Unfortunately that key idea runs counter to the entire django/requirements/pypi architecture of modular design, which is in place so projects don't become monoliths that require knowledge of all features. In order for the Core CMS to be able to grow dynamically, it needs to remain (as much as possible) feature agnostic--so it doesn't have references to features/plugins/apps unless they are universally required (like the auth functionality we're building in here). 

In the same way that Django need-not know about django-cms, nor django-cms about each plugin that's built for it, a modular architecture means we build plugins/apps that can be added to Core CMS via a config like requirements and settings.py (installed_apps, etc) and not via if/else conditionals.

This is the same pattern followed throughout all software design, and without it, software like Django would quickly become unmanageable, which is where we're headed with Core CMS if we don't start moving towards adopting the standard model and pattern that makes software flexible and reusable.. and that means the core cms code should not be aware of any "features" unless they are deemed as required for most if-not-all use cases, and even then, it may make sense to break out some of those if they become overly large or useful on their own.

Reply:
[Private discussion](https://tacc-team.slack.com/archives/G015BTGGLF5/p1631650173051100) has lead to agreement on the "Within taccsite_custom" solution—specifically `taccsite_custom:texascale-cms/`.

<details><summary>Reasons</summary>

- We need not add _all_ functionality to Core.
- Ideally, we have one implementation via a TACC blog plugin that overrides `djangocms-blog` only as necessary.
- Currently, only Texascale is using the blog. We can commit reference implementation in its `settings_custom.py`.

</details>

<details><summary>Drawbacks</summary>

- There is not one source of code (yet).
- Other domains do not know how to add blog.
    - I'll use documentation as tape.

</details>